File: python/2379.Minimum-Recolors-to-Get-K-Consecutive-Black-Blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        count_w , min_w = 0, 0

        for i in range(k):
            if blocks[i] == 'W':
                count_w += 1

        min_w = count_w

        for i in range(k, len(blocks)):
            if blocks[i - k] == 'W':
                count_w -= 1
            if blocks[i] == 'W':
                count_w += 1
            min_w = min(min_w, count_w)

        return min_w

File: javascript/2379.Minimum-Recolors-to-Get-K-Consecutive-Black-Blocks.js
/**
 * @param {string} blocks
 * @param {number} k
 * @return {number}
 */
var minimumRecolors = function (blocks, k) {
  let countW = 0;
  let min = 0;

  for (let i = 0; i < k; i++) {
    if (blocks[i] === "W") countW += 1;
  }

  min = countW;

  for (let i = k; i < blocks.length; i++) {
    if (blocks[i - k] === "W") countW -= 1; // outgoing
    if (blocks[i] === "W") countW += 1; // incoming

    min = Math.min(min, countW);
  }

  return min;
};
